### PR TITLE
 metric alert migration

### PR DESCRIPTION
**Pull request by SINIKI**
## Description

Updated metric alert controls logic to support classic alert migration.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.Framework/Abstracts/SVTBase.ps1 [hunk 1]
```diff
@@ -25,6 +25,7 @@ class SVTBase: AzSKRoot
 	[string] $PartialScanIdentifier = [string]::Empty
 	[ComplianceStateTableEntity[]] $ComplianceStateData = @();
 	[PSObject[]] $ChildSvtObjects = @();
+	[PSObject[]] $MetricAlerts = @();
```
Comment (VishalHaibatpure) [inline src/AzSK.Framework/Abstracts/SVTBase.ps1]: why are we adding MetricAlerts at base class property. Its only been used in one method.

File: src/AzSK.Framework/Abstracts/SVTBase.ps1 [hunk 2]
```diff
@@ -1202,10 +1213,83 @@ class SVTBase: AzSKRoot
...
+			$resourceAlerts = @()
+            # get classic alerts
+			$resourceAlerts += (Get-AzAlertRule -ResourceGroup $this.ResourceContext.ResourceGroupName -WarningAction SilentlyContinue) |
 								Where-Object { $_.Condition -and $_.Condition.DataSource } |
 								Where-Object { $_.Condition.DataSource.ResourceUri -eq $resId };
 
+			# get non-classic alerts
+            try
+            {
+                $apiURL = "https://management.azure.com/subscriptions/{0}/providers/Microsoft.Insights/metricAlerts?api-version=2018-03-01" -f $($this.SubscriptionContext.SubscriptionId)
```
Comment (VishalHaibatpure) [inline src/AzSK.Framework/Abstracts/SVTBase.ps1]: Fetch resource level metric alerts only

Comment (SINIKI) [reply src/AzSK.Framework/Abstracts/SVTBase.ps1]: Updated api to fetch alert at resource group level

File: src/AzSK.Framework/Abstracts/SVTBase.ps1 [hunk 3]
```diff
@@ -1202,10 +1202,83 @@ class SVTBase: AzSKRoot
...
+            try
+            {
+                $apiURL = "https://management.azure.com/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.Insights/metricAlerts?api-version=2018-03-01" -f $($this.SubscriptionContext.SubscriptionId), $this.ResourceContext.ResourceGroupName
+                $v2Alerts = [WebRequestHelper]::InvokeGetWebRequest($apiURL)                
+                $v2Alerts = $v2Alerts | Where-Object { $_.properties.scopes -contains $resId }
+                if(($v2Alerts |  Measure-Object).Count -gt 0)
+                {
+                    $v2Alerts |  ForEach-Object {
+						if([Helpers]::CheckMember($_,"properties"))
+						{
```
Comment (VishalHaibatpure) [inline src/AzSK.Framework/Abstracts/SVTBase.ps1]: Need optimization for object comparision. Refer how we do ASC configuratin or any previous metric/diagnotistic checks

**Pull request merged**